### PR TITLE
Feat/2 7 1 env schema

### PR DESCRIPTION
**Pull request by jonahkeegan**
# Pull Request - CritGenius: Listener

## Overview

**Brief description of changes:**

**Related Issue(s):**

**Type of Change:**

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as
      expected)
- [ ] Performance improvement
- [ ] Documentation update
- [ ] Refactoring (no functional changes)
- [ ] Security enhancement
- [ ] Infrastructure/DevOps changes

### 🎙️ Audio Capture & Processing

- [ ] **Web Audio API Integration:** Changes maintain compatibility with target browsers (Chrome
      70+, Firefox 65+, Safari 12+, Edge 79+)
- [ ] **Microphone Permissions:** Proper handling of user permission requests and denials
- [ ] **Audio Quality:** Maintains 16kHz sample rate, mono channel, with noise suppression and echo
      cancellation
- [ ] **Buffer Management:** Audio chunks processed efficiently without memory leaks or overflow
- [ ] **Error Handling:** Graceful handling of audio device errors, disconnections, and browser
      limitations

### ⚡ Real-Time Performance Requirements

- [ ] **Latency Target:** End-to-end processing maintains <500ms latency (measured and documented)
- [ ] **Memory Usage:** No memory leaks during extended sessions (tested for >2 hours)
- [ ] **CPU Impact:** Changes don't cause UI blocking or excessive CPU usage
- [ ] **WebSocket Performance:** Real-time data streaming remains stable under load
- [ ] **Concurrent Sessions:** Tested with multiple simultaneous audio sessions if applicable

### 🗣️ AssemblyAI Integration & Transcription

- [ ] **API Integration:** Proper error handling for AssemblyAI service interruptions
- [ ] **Speaker Diarization:** Speaker identification accuracy maintained (>90% target)
- [ ] **Transcription Quality:** D&D terminology recognition preserved (>95% accuracy target)
- [ ] **Rate Limiting:** Respects AssemblyAI API limits and implements appropriate backoff
- [ ] **Real-Time Streaming:** Maintains stable connection with AssemblyAI real-time service

### 📝 TypeScript & Code Quality

- [ ] **Type Safety:** All new code properly typed with TypeScript strict mode compliance
- [ ] **ESLint:** No linting errors (run `npm run lint`)
- [ ] **Prettier:** Code formatted consistently (run `npm run format`)
- [ ] **Code Comments:** Complex audio processing logic documented with clear comments
- [ ] **API Documentation:** New endpoints or changes documented in appropriate files

### 🧪 Testing Requirements

- [ ] **Unit Tests:** New functionality covered by unit tests (Jest + Testing Library)
- [ ] **Integration Tests:** Audio processing pipeline tested end-to-end
- [ ] **Cross-Browser Testing:** Tested in Chrome, Firefox, Safari, and Edge (if UI changes)
- [ ] **Performance Tests:** Latency and resource usage measured and within targets
- [ ] **Error Scenarios:** Edge cases and error conditions tested
- [ ] **Test Coverage:** Maintains minimum 90% code coverage

### 🌐 Browser & Device Compatibility

- [ ] **Responsive Design:** Changes work across desktop and mobile viewports
- [ ] **Progressive Enhancement:** Graceful degradation when Web Audio API unavailable
- [ ] **Device Testing:** Tested with different microphone setups and audio devices
- [ ] **Network Conditions:** Tested under various network speeds and stability
- [ ] **Browser Console:** No console errors or warnings in supported browsers

### 🔒 Privacy-First Architecture

- [ ] **Audio Data Handling:** No unnecessary audio data storage or transmission
- [ ] **User Consent:** Clear user consent flows for microphone access and data processing
- [ ] **Data Encryption:** Sensitive data encrypted in transit and at rest
- [ ] **API Keys:** No hardcoded API keys or sensitive credentials
- [ ] **CORS Configuration:** Proper cross-origin resource sharing configuration
- [ ] **Input Validation:** All user inputs properly validated and sanitized

### 🛡️ Security Best Practices

- [ ] **Dependency Updates:** No known security vulnerabilities in dependencies
- [ ] **HTTPS/WSS:** All communication uses encrypted connections
- [ ] **Content Security Policy:** CSP headers configured appropriately
- [ ] **Authentication:** Proper authentication and authorization if applicable
- [ ] **Rate Limiting:** Protection against abuse and DoS attacks

### 🎭 Character & Speaker Mapping

- [ ] **Voice Profiles:** Speaker identification and character mapping functionality preserved
- [ ] **Session Persistence:** Character assignments persist across session interruptions
- [ ] **Multi-Player Support:** Changes support multiple players and characters
- [ ] **DM Tools:** Dungeon Master specific features maintained and enhanced
- [ ] **Campaign Integration:** Compatibility with ongoing campaign session management

### 📚 D&D Terminology & Context

- [ ] **Game Terms:** Proper handling of D&D-specific vocabulary (spells, abilities, locations)
- [ ] **Dice Notation:** Recognition of dice rolls and game mechanics terminology
- [ ] **Character Names:** Proper handling of fantasy character and location names
- [ ] **Session Context:** Maintains narrative continuity and context awareness
- [ ] **Export Formats:** Compatible with popular D&D tools and platforms

### 📖 Documentation Updates

- [ ] **README:** Updated if installation, setup, or usage instructions changed
- [ ] **API Documentation:** New endpoints or changes documented
- [ ] **CHANGELOG:** Entry added for user-facing changes
- [ ] **Code Comments:** Complex logic explained with clear comments
- [ ] **Deployment Guide:** Updated if deployment process changes

### 💬 Accessibility & User Experience

- [ ] **Screen Reader Support:** Changes accessible via keyboard and screen readers
- [ ] **Keyboard Navigation:** All interactive elements accessible via keyboard
- [ ] **Error Messages:** Clear, actionable error messages for users
- [ ] **Loading States:** Appropriate loading indicators for async operations
- [ ] **User Feedback:** Clear success/failure feedback for user actions

### 🔗 System Integration

- [ ] **API Compatibility:** Changes maintain compatibility with other CritGenius components
- [ ] **Data Formats:** Export formats compatible with Prompter, LLM, and Publisher modules
- [ ] **Webhook Integration:** Event notifications sent to appropriate ecosystem services
- [ ] **Version Compatibility:** Changes maintain backward compatibility where possible
- [ ] **Migration Path:** Clear upgrade path for existing users if breaking changes

### 📊 Performance Metrics

**Latency Measurements:**

- End-to-end audio processing: \_\_\_ms (target: <500ms)
- UI responsiveness: \_\_\_ms (target: <100ms)
- WebSocket round-trip: \_\_\_ms

**Resource Usage:**

- Memory usage after 2-hour session: \_\_\_MB
- CPU usage during peak processing: \_\_\_%
- Network bandwidth utilization: \_\_\_KB/s

**Accuracy Metrics:**

- Transcription accuracy on test dataset: \_\_\_%
- Speaker identification accuracy: \_\_\_%
- D&D terminology recognition rate: \_\_\_%

### 🔍 Browser Testing Results

- [ ] Chrome (latest): Tested and working
- [ ] Firefox (latest): Tested and working
- [ ] Safari (latest): Tested and working
- [ ] Edge (latest): Tested and working

**Device Testing:**

- [ ] Desktop/laptop microphones
- [ ] USB headset microphones
- [ ] Bluetooth audio devices
- [ ] Mobile device testing (if applicable)

### 🚀 Production Readiness

- [ ] **Environment Variables:** New configuration properly documented in .env.example
- [ ] **Database Migrations:** Schema changes include proper migration scripts
- [ ] **Feature Flags:** New features behind feature flags if needed
- [ ] **Rollback Plan:** Clear rollback procedure documented
- [ ] **Monitoring:** Appropriate logging and monitoring added for new features

### ⚙️ CI/CD Pipeline

- [ ] **Build Success:** All CI/CD checks passing
- [ ] **Docker Compatibility:** Changes work in containerized environment
- [ ] **Configuration Management:** Environment-specific configurations handled properly
- [ ] **Health Checks:** New endpoints include appropriate health checks
- [ ] **Performance Baseline:** New performance benchmarks established if needed

### ✅ Final Verification

- [ ] **Code Review:** At least one thorough code review completed
- [ ] **Manual Testing:** Manually tested in development environment
- [ ] **Integration Testing:** Tested with full application stack
- [ ] **Documentation Review:** All documentation updates reviewed and accurate
- [ ] **Breaking Changes:** Breaking changes documented and communicated
- [ ] **Release Notes:** Contribution to release notes prepared if needed

## Additional Notes

<!-- Any additional context, concerns, or considerations

### CODE REVIEW

File: packages/server/src/realtime/sessionManager.ts [hunk 1]
```diff
@@ -87,7 +87,7 @@ export class SessionManager {
 
     // Basic format validation to fail fast on obviously invalid keys
     const looksValid =
-      resolvedKey.length >= 32 && !/\s/.test(resolvedKey);
+      resolvedKey.length >= 32 && /^[a-f0-9]{32,}$/i.test(resolvedKey);
```
[the remark this answers is not in the document]
Comment (jonahkeegan) [reply packages/server/src/realtime/sessionManager.ts:90]: Fixed in local codebase.
